src/Kernel/client/telemetry.ts

```diff
@@ -148,18 +152,19 @@ class TelemetryHelper {
             const body = await response.json();
             return body;
         } catch (ex) {
+            console.error(`Error fetching url ${request.toString()}`)
             throw ex;
         }
     }
 
-    private async getClientInfoAsync(setHasConsent: boolean = false): Promise<ClientInfo> {
+    private async getClientInfoAsync(setHasConsent = false): Promise<ClientInfo> {
         console.log("Getting ClientInfo");
-        var url =
+        const url =
             CLIENT_INFO_API_URL
             + (setHasConsent ? "&hasconsent=1" : "")
-            + ((this.origin != null && this.origin != "") ? "&origin=" + this.origin : "");
+            + ((this.origin !== null && this.origin !== "") ? "&origin=" + this.origin : "");
         try {
-            var clientInfo = await this.fetchExt<ClientInfo>(url);
+            const clientInfo = await this.fetchExt<ClientInfo>(url);
```

Comment:
From what I can tell, this GetClientInfo API returns `HasConsent: true` even if I have no cookies set locally. I presume that is because my request is coming from the US? What's the best way to test this locally... is there some URL parameter I can add to the request to override the country code?

Reply:
This is kind of hard to test in an automated way. The Microsoft cookie API has all the logic in it and it about when it shows or not the cookie banner and response if the user has consented, and it could change in the future becoming more or less conservative depending on the user's country (via reverse IP) and locale (browser's primary language).

The way I tested it was to set the "ENABLE_CLIENT_TELEMETRY" and use a VPN to emulate traffic coming from Belgium, for example.

Reply:
Agreed, we wouldn't want automated tests that actually rely on the behavior of the cookie API.

One idea; Could we have an override here (e.g., maybe via another special `hostingEnvironment` value, or perhaps something else similar) that skips the call to the service and always uses some hard-coded `ClientInfo` object which has `HasConsent: false`?

After our discussion, I understand that we can't easily get automated tests to actually show the banner, since that code is outside our control. But it could be useful to at least allow us to exercise our own portion of the code.

Reply:
As discussed, unit tests can be added in a separate PR, since this is only a modification to pre-existing code. It would be great if you could create an issue to track adding unit tests for this code.

Reply:
Created issue #370 to add unit tests for this and other typescript code.